node-ui/src/hooks/useIcp.tsx

```diff
@@ -189,6 +190,10 @@ export function useIcp(): useIcpReturn {
 
   const signMessageAndLogin = useCallback(
     async (isLogin: boolean, setErrorMessage: (msg: string) => void) => {
+      if (!getAppEndpointKey()) {
+        navigate('/?node_url_error=true');
```

Comment:
why don't you just navigate to setup page?

Reply:
Because its a bad UX redirecting user and not knowing why you got redirected. You press sign message and then youre back at setup again, would look like its a bug

Reply:
But how can local storage lose the necessary data? There must be action before it.

Reply:
> This is the only way that this can be recreated (by clearning local storage) in other examples it will give you error under login card of each network.

Reply:
Removed popup and redirect back to setup page.